src/material/schematics/ng-generate/m3-theme/index.ts

```diff
@@ -42,49 +44,82 @@ const NEUTRAL_HUE_TONES = [...HUE_TONES, ...NEUTRAL_HUES.keys()];
 
 /**
  * Gets color tonal palettes generated by Material from the provided color.
- * @param color Color that represent primary to generate all the tonal palettes.
- * @returns Object with tonal palettes for each color
+ * @param primaryPalette Tonal palette that represents primary.
+ * @param secondaryPalette Tonal palette that represents secondary.
+ * @param tertiaryPalette Tonal palette that represents tertiary.
+ * @param neutralPalette Tonal palette that represents neutral.
+ * @param neutralVariantPalette Tonal palette that represents neutral variant.
+ * @param isDark Boolean to represent if the scheme is for a dark or light theme.
+ * @param contrastLevel Number between -1 and 1 for the contrast level. 0 is the standard contrast
+ * and 1 represents high contrast.
+ * @returns Dynamic scheme for provided theme and contrast level
  */
-function getMaterialTonalPalettes(color: string): {
-  primary: TonalPalette;
-  secondary: TonalPalette;
-  tertiary: TonalPalette;
-  neutral: TonalPalette;
-  neutralVariant: TonalPalette;
-  error: TonalPalette;
-} {
-  try {
-    let argbColor = argbFromHex(color);
-    const scheme = new SchemeContent(
-      Hct.fromInt(argbColor),
-      false, // Tonal palettes are the same for light and dark themes
-      0.0,
-    );
+function getMaterialDynamicScheme(
+  primaryPalette: TonalPalette,
+  secondaryPalette: TonalPalette,
+  tertiaryPalette: TonalPalette,
+  neutralPalette: TonalPalette,
+  neutralVariantPalette: TonalPalette,
+  isDark: boolean,
+  contrastLevel: number,
+): DynamicScheme {
+  return new DynamicScheme({
+    sourceColorArgb: primaryPalette.keyColor.toInt(),
+    variant: 6, // Variant enum is not accessible outside of @material/material-color-utilities
```

Comment:
What is "6"? Should include this in comments

Reply:
Updated the comment, variant is supposed to take a Variant enum but it isn't exposed outside the package